app/prioritizer.go

```diff
@@ -0,0 +1,186 @@
+package app
+
+import (
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	cosmosante "github.com/cosmos/cosmos-sdk/x/auth/ante"
+	paramskeeper "github.com/cosmos/cosmos-sdk/x/params/keeper"
+	upgradekeeper "github.com/cosmos/cosmos-sdk/x/upgrade/keeper"
+	"github.com/ethereum/go-ethereum/consensus/misc/eip4844"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/sei-protocol/sei-chain/app/antedecorators"
+	"github.com/sei-protocol/sei-chain/utils"
+	evmante "github.com/sei-protocol/sei-chain/x/evm/ante"
+	"github.com/sei-protocol/sei-chain/x/evm/derived"
+	evmkeeper "github.com/sei-protocol/sei-chain/x/evm/keeper"
+	evmtypes "github.com/sei-protocol/sei-chain/x/evm/types"
+	oracletypes "github.com/sei-protocol/sei-chain/x/oracle/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+var _ sdk.TxPrioritizer = (*SeiTxPrioritizer)(nil).GetTxPriorityHint
+
+type SeiTxPrioritizer struct {
+	evmKeeper     *evmkeeper.Keeper
+	upgradeKeeper *upgradekeeper.Keeper
+	paramsKeeper  *paramskeeper.Keeper
+	logger        log.Logger
+}
+
+func NewSeiTxPrioritizer(logger log.Logger, ek *evmkeeper.Keeper, uk *upgradekeeper.Keeper, pk *paramskeeper.Keeper) *SeiTxPrioritizer {
+	return &SeiTxPrioritizer{
+		logger:        logger,
+		evmKeeper:     ek,
+		upgradeKeeper: uk,
+		paramsKeeper:  pk,
+	}
+}
+
+func (s *SeiTxPrioritizer) GetTxPriorityHint(ctx sdk.Context, tx sdk.Tx) (_priorityHint int64, _err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			// Fall back to no-op priority if we panic for any reason. This is to avoid DoS
+			// vectors where a malicious actor crafts a transaction that panics the
+			// prioritizer. Since the prioritizer is used as a hint only, it's safe to fall
+			// back to zero priority in this case and log the panic for monitoring purposes.
+			s.logger.Error("tx prioritizer panicked. Falling back on no priority", "error", r)
+			_priorityHint = 0
+			_err = nil
+		}
+	}()
+	if ctx.HasPriority() {
+		// The context already has a priority set, return it.
+		return ctx.Priority(), nil
+	}
+
+	if ok, err := evmante.IsEVMMessage(tx); err != nil {
+		return 0, err
+	} else if ok {
+		evmTx := evmtypes.GetEVMTransactionMessage(tx)
+		if evmTx != nil {
+			return s.getEvmTxPriority(ctx, evmTx)
```

Comment:
Do we need to update the context by calling WithPriority after we get the priority?

Reply:
Nope. Left over from back when get priority wasn't meant to be a hint. Thank you for pointing that out

Reply:
gotcha, make sense